packages/app/src/components/Dimensions/DimensionOptions.js

```diff
@@ -1,61 +1,111 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
+import i18n from '@dhis2/d2-i18n';
 import MenuItem from '@material-ui/core/MenuItem';
+import OptionsButton from '../DimensionOptions/OptionsButton';
 import DropDown from './DropDown';
-import MoreHorizontalIcon from '../../assets/MoreHorizontalIcon';
 import {
     acAddUiLayoutDimensions,
-    acSetUiItems,
     acRemoveUiLayoutDimensions,
     acSetUiActiveModalDialog,
 } from '../../actions/ui';
 import { sGetUiLayout, sGetUiItems } from '../../reducers/ui';
-
-import { ADD_TO_LAYOUT_OPTIONS } from '../../modules/layout';
+import { axisLabels } from '../../modules/layout';
+import { isYearOverYear } from '../../modules/chartTypes';
 import { styles } from './styles/DimensionOptions.style';
 
-export const OptionsButton = ({ action }) => {
-    return (
-        <button style={styles.dropDownButton} onClick={action} tabIndex={1}>
-            <MoreHorizontalIcon />
-        </button>
-    );
-};
-
+const FILTER = 'filters';
 export class DimensionOptions extends Component {
     state = { anchorEl: null };
 
-    handleClick = event => {
-        event && event.stopPropagation();
+    onOpenMenu = event => {
         this.setState({ anchorEl: event.currentTarget });
     };
 
-    handleClose = event => {
-        event && event.stopPropagation();
+    onCloseMenu = () => {
         this.setState({ anchorEl: null });
-        this.props.onClose();
+        this.props.onCloseMenu();
     };
 
     addDimension = axisName => {
+        this.onCloseMenu();
         this.props.onAddDimension({ [this.props.id]: axisName });
-        this.handleClose();
-        setTimeout(() => this.props.openDialog(this.props.id), 10);
+
+        const items = this.props.items[this.props.id];
+        const hasNoItems = Boolean(!items || !items.length);
+
+        if (hasNoItems) {
+            this.props.openDialog(this.props.id);
+        }
+    };
+
+    removeDimension = id => {
+        this.onCloseMenu();
+        this.props.removeDimension(id);
     };
 
+    getAddToItems = () => [
+        isYearOverYear(this.props.type)
+            ? this.renderMenuItem(
+                  `add-to-${this.props.id}`,
+                  FILTER,
+                  this.addDimension,
+                  `Add to ${axisLabels[FILTER]}`
+              )
+            : Object.entries(axisLabels).map(([key, label]) =>
+                  this.renderMenuItem(
+                      `add-to-${key}`,
+                      key,
+                      this.addDimension,
+                      `${i18n.t('Add to')} ${label}`
```

Comment:
For this type of strings, is best to use variables within the string, as different languages might need to put the label part in a different place in the string.
Use this format instead: `${i18n.t('Add to {label}', { label })`.